### PR TITLE
Fix #24809

### LINKED ISSUE #24809: Mdx integration broken since Gatsby 2.22.21
## Description

Since gatsby `2.22.21` the mdx integration seems to be broken.
The only thing changed between `2.22.20` and `2.22.21` is this commit (+ gatsby-admin according to lock file): https://github.com/gatsbyjs/gatsby/commit/64858b12b34a3fab45b1bb06a6b65803231569e2

-> `gatsby develop`: ✅ 
-> `gatsby build`: broken

### Steps to reproduce

1. Open Sandbox with Gatsby Starter + mdx (https://codesandbox.io/s/elegant-babbage-z0g4m?file=/gatsby-config.js):

```sh
yarn add gatsby-plugin-mdx @mdx-js/mdx @mdx-js/react
```

```js
// gatsby-config.js
...
{
      resolve: `gatsby-source-filesystem`,
      options: {
        name: `pages`,
        path: `${__dirname}/src/pages/`
      }
},
`gatsby-plugin-mdx`
...
```

2. Run `yarn build` (`gatsby build`): Success
3. Change dependency `gatsby` from `2.22.20` to `2.22.21`
4. Run `yarn build` (`gatsby build`): Error

```sh
success createPages - 0.003s

ERROR #11328

A page component must export a React component for it to be valid. Please make sure this file exports a React component:

undefined

not finished createPagesStatefully - 0.203s
error Command failed with exit code 1.
```

### Expected result

- Run `gatsby build` successfully

### Actual result

```
// gatsby build

...
not finished createPagesStatefully
error Command failed with exit code 1.
```

### Environment

```
  System:
    OS: Linux 4.19 Debian GNU/Linux 10 (buster) 10 (buster)
    CPU: (12) x64 Intel(R) Core(TM) i7-8700 CPU @ 3.20GHz
    Shell: 5.0.3 - /bin/bash
  Binaries:
    Node: 10.20.1 - /tmp/yarn--1591357611412-0.21233396036717855/node
    Yarn: 1.22.4 - /tmp/yarn--1591357611412-0.21233396036717855/yarn
    npm: 6.14.4 - /usr/local/bin/npm
  Languages:
    Python: 2.7.16 - /usr/bin/python
  npmPackages:
    gatsby: 2.22.21 => 2.22.21
    gatsby-image: 2.4.6 => 2.4.6
    gatsby-plugin-manifest: 2.4.10 => 2.4.10
    gatsby-plugin-mdx: 1.2.14 => 1.2.14
    gatsby-plugin-offline: 3.2.8 => 3.2.8
    gatsby-plugin-react-helmet: 3.3.3 => 3.3.3
    gatsby-plugin-sharp: 2.6.10 => 2.6.10
    gatsby-source-filesystem: 2.3.10 => 2.3.10
    gatsby-transformer-sharp: 2.5.4 => 2.5.4
```

### PR DESCRIPTION
**Pull request by iamvishnusankar**
Fix #24809

### CODE REVIEW

**Comment by ascorbic** [comment]
Nice catch!

**Comment by laurieontech** [comment]
Looks good! Appreciate the fix.

File: packages/gatsby/src/utils/validate-page-component.ts [hunk 1]
```diff
@@ -79,11 +79,13 @@ export function validatePageComponent(
...
-      fileContent.match(/export \{.* as default.*\}/s) ||
-      // this check only applies to js and ts, not mdx
-      /\.(jsx?|tsx?)/.test(path.extname(component))
+      fileContent.match(/export \{.* as default.*\}/s)
 
-    if (!includesDefaultExport) {
+    // this check only applies to js and ts, not mdx
+    if (
+      /\.(jsx?|tsx?)/.test(path.extname(component)) &&
+      !includesDefaultExport
```
Comment (muescha) [inline packages/gatsby/src/utils/validate-page-component.ts:87]: did it speedup;
- when at first the `(jsx?|tsx?)/.test` is done 
- and then only if it match the calculation for `includesDefaultExport` with the many `manytext.includes` are calculated ?

Comment (iamvishnusankar) [reply packages/gatsby/src/utils/validate-page-component.ts:87]: @muescha Yes, that's correct. Performing `includesDefaultExport` before extension check is unnecessary.  Fixed it with PR #24837

**Comment by pvdz** [comment]
(Thanks for this fix. Please consider a more descriptive title next time :) )

**Comment by kentcdodds** [comment]
Would love for this to be released 🙏

**Comment by LekoArts** [comment]
@kentcdodds We released `gatsby@2.23.2` :)

**Pull request merged**